### PR TITLE
Shooter control

### PR DESCRIPTION
**Pull request by OliverW10**
shooter controller recives wants_to_fire from robot.py or auto, when its flywheels are at speed and its aiming at the target it sets wants_to_fire on indexer which will feed the ball if we have one

### CODE REVIEW

File: autonomous/autonomous.py [hunk 1]
```diff
@@ -83,27 +83,33 @@ def generate_trap_profile(
...
-        return (
+        ret = (
             TrapezoidProfile(
                 self.linear_constraints,
                 goal=TrapezoidProfile.State(stop_point, 0),
                 initial=current_state,
             ),
             stop_point,
         )
+        print(f"[{self.MODE_NAME}] generated {ret[0].totalTime()}s")
```
Comment (auscompgeek) [inline autonomous/autonomous.py:94]: Please use `self.logger` or remove the print calls.

File: components/intake.py [hunk 2]
```diff
@@ -11,6 +11,7 @@ class Intake:
     speed = magicbot.will_reset_to(0.0)
 
     def setup(self) -> None:
+        self.intake_motor.restoreFactoryDefaults()
```
Comment (auscompgeek) [inline components/intake.py:14]: This seems unrelated to shooter?

Comment (OliverW10) [reply components/intake.py:14]: yes, the intake would sometimes run backwards and this seemed to fix it, should i make a seperate pull request?

Comment (auscompgeek) [reply components/intake.py:14]: A separate PR would be good. It seems like we should just always call `setInverted` on our motor controllers though?

File: autonomous/autonomous.py [hunk 3]
```diff
@@ -50,7 +50,7 @@ def __init__(self):
         # how far around the current position is used to smooth the path
         self.look_around = 0.3
 
-        print(f"[{self.MODE_NAME}] total length {self.total_length}s")
+        self.logger.info(f"[{self.MODE_NAME}] total length {self.total_length}s")
```
Comment (auscompgeek) [inline autonomous/autonomous.py:53]: Isn't MODE_NAME redundant here?
```suggestion
        self.logger.info(f"total length {self.total_length}s")
```

File: autonomous/autonomous.py [hunk 4]
```diff
@@ -83,27 +83,33 @@ def generate_trap_profile(
...
-        return (
+        ret = (
             TrapezoidProfile(
                 self.linear_constraints,
                 goal=TrapezoidProfile.State(stop_point, 0),
                 initial=current_state,
             ),
             stop_point,
         )
+        self.logger.info(f"[{self.MODE_NAME}] generated {ret[0].totalTime()}s")
```
Comment (auscompgeek) [inline autonomous/autonomous.py:94]: Ditto.
```suggestion
        self.logger.info(f"generated {ret[0].totalTime()}s")
```

File: autonomous/autonomous.py [hunk 5]
```diff
@@ -83,27 +83,33 @@ def generate_trap_profile(
...
         linear_state = self.trap_profile.calculate(trap_time)
-        is_done = self.trap_profile.isFinished(trap_time)
+        is_done = self.trap_profile.isFinished(
+            trap_time
+        )  # TODO: change to our error is below value
 
         if is_done:
-            print(f"[{self.MODE_NAME}] Done at {tm}")
-            self.done()
+            self.logger.info(f"[{self.MODE_NAME}] Done at {tm}")
```
Comment (auscompgeek) [inline autonomous/autonomous.py:111]:
```suggestion
            self.logger.info(f"Done at {tm}")
```

File: autonomous/autonomous.py [hunk 6]
```diff
@@ -83,27 +83,33 @@ def generate_trap_profile(
...
         # always be trying to fire
-        # self.shooter_control.fire_input()
+        self.shooter_control.wants_to_fire = True
         # calculate speed and position from current trapazoidal profile
         trap_time = tm
         linear_state = self.trap_profile.calculate(trap_time)
-        is_done = self.trap_profile.isFinished(trap_time)
+        is_done = self.trap_profile.isFinished(
+            trap_time
+        )  # TODO: change to our error is below value
```
Comment (auscompgeek) [inline autonomous/autonomous.py:108]: Please put the comment before the line it's about.

File: controllers/indexer.py [hunk 7]
```diff
@@ -89,4 +86,10 @@ def trapping(self, state_tm) -> None:
             self.indexer.set(1, 1)
 
     def check_firing(self) -> bool:
-        return self.wants_to_fire and (self.indexer.has_back() or self.has_back)
+        """Checks if we want to be firing right now"""
```
Comment (auscompgeek) [inline controllers/indexer.py:89]: nit: maybe call this method `should_fire`?

**Review by auscompgeek — APPROVED** [review]
lgtm with comments

File: autonomous/autonomous.py (hunk 3 above)
Comment (OliverW10) [reply autonomous/autonomous.py:53]: all auto's get initialized so if you wanted to log their lengths you'd want a way to distinguish then, however we probably don't need to know the lengths

Comment (auscompgeek) [reply autonomous/autonomous.py]: :thinking: doesn't the logger show your mode's name already?

Comment (OliverW10) [reply autonomous/autonomous.py]: yes, hadnt looked since it was changed from print to logger
